src/FixedPointNumbers.jl

```diff
@@ -40,54 +41,87 @@ export
     # Functions
     scaledual
 
-reinterpret(x::FixedPoint) = x.i
+getindex(x::AbstractFixedPoint) = x.i
```

Comment:
The problem, however, with your solution is that it conflicts with
```jl
julia> 5[]
5

julia> 3.2[]
3.2
```
So anyone who writes code like
```jl
function mysum(X)
    s = zero(first(X))
    for x in X
        s += x
    end
    s
```
will get a very surprising answer from `mysum(0xffuf8)`.

`x.i` might be used inside `FixedPointNumbers`, but I've never seen such usage outside. If this is intended only for internal use, rather than overload an exported function from Base you could define `raw` (or something) as an internal function.